### ISSUE TITLE
mWeb - Security- New Password field is hidden by Save button

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#9620]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Precondition: user has password suggestion in her/his phone
1. Go to staging.new.expensify.com
2. Log in with any account
3. Go to Setting - Security
4. Put Incorrect password on Current Password field
5. Tap on New password field

## Expected Result:
User is able to see a new Password field

## Actual Result:
New Password field is hidden by Save button

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.79.2

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/176320870-60058ea7-d593-4f6d-91fc-cdc89515e96b.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by stitesExpensify** [comment]
It looks like the problem is actually the keyboard popping up, not the password suggestions (it's hidden behind the suggestions here). Looks like we need a `<keyboardAvoidingView>` somewhere. Should be an easy fix

**Comment by jliexpensify** [comment]
Posted:

- Internal: https://www.upwork.com/ab/applicants/1542366198684119040/job-details
- External: https://www.upwork.com/jobs/~01276b1b1746a62a00

I tested this on Android (same version) and it wasn't an issue for me, so perhaps it might be affecting iOS?

**Comment by muaazsheikh** [comment]
<KeyboardAvoidingView behavior='padding'> it will be work we need to more changes in parent view using flex

**Comment by jliexpensify** [comment]
@muaazsheikh - please feel free to submit a proposal for review!

**Comment by muaazsheikh** [comment]
@jliexpensify - i already did can you please review it.

**Comment by jliexpensify** [comment]
@muaazsheikh - it looks like you made a proposal in your Cover Letter in Upworks.

Can you share your proposal within this GH instead? Our process is to have contributors post proposals in the GH for review by our team. Thanks in advance!

**Comment by Santhosh-Sellavel** [comment]
Waiting for proposal here

**Comment by muaazsheikh** [comment]
@jliexpensify  
There is muliple Ways to handle button upside over keyboard there is a way to make whole screen is KeyboardAvoidingviee or one more third party in react native which is KeyboardAwareScrollView if both will not work then we can manage through flex, i was facing this issue number of times but yeah it can be handled.

**Comment by Santhosh-Sellavel** [comment]
@muaazsheikh 

Can you elaborate more and write a detailed/informative on the code changes you are proposing. 

Look through [these](https://github.com/Expensify/App/issues?q=is%3Aissue+is%3Aclosed+label%3A%22Awaiting+Payment%22) past issues to get an idea about how we expect proposals.

If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) first!

Thanks!

**Comment by damdafayton** [comment]
@kbecciv @muaazsheikh I think its bad css. Probably save button is positioned absolute/relative to the bottom and keyboard is pushing it up.

**Comment by damdafayton** [comment]
Ok I have solved it.
You need to add `min-height: 100vh;` to the element with this class `css-1dbjc4n r-13awgt0` which is the container of the button. Then it wont move up any more.

**Comment by damdafayton** [comment]
@jliexpensify

**Comment by damdafayton** [comment]
I am trying to fix it to open PR but Im unable to run the app on local.

`npm run web-server` gives this error: `..expensify-bug\node_modules\expand-tilde\index.js'. Please verify that the package.json has a valid "main" entry`.

If you can help me setup on local I can give the solution fastly.

**Comment by stitesExpensify** [comment]
Please follow the instructions here to get started as a contributor 😄

https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md

**Comment by damdafayton** [comment]
I have noticed that fixed button at the bottom of the screen is a design choice for the whole app. So mentioned problem doesnt exist on android. It must be related to ios keyboard settings.

**Comment by Santhosh-Sellavel** [comment]
The issue is still reproducible

https://user-images.githubusercontent.com/85645967/178160238-352ae793-51a2-423a-9686-7cea40ba306d.mov

**Comment by damdafayton** [comment]
> The issue is still reproducible
> 
>  Screen.Recording.2022-07-11.at.1.29.34.AM.mov

This is different than the problem in the initial post. 

On initial post button is pushed up by keyboard. In your case button is below keyboard.

**Comment by damdafayton** [comment]
Decision-makers of this site should decide and inform developers wether they want the buttons to be stable on the page or move with the keyboard. 

And in your case problem is page not scrolling up automatically.

**Comment by Santhosh-Sellavel** [comment]
> The issue is still reproducible
> 
>  Screen.Recording.2022-07-11.at.1.29.34.AM.mov
Oops, I mistook the issue. The Sign In Button is partially hidden above that's a different issue.

In this issue, we are discussing the `New password` field, This issue is no longer reproducible.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by jliexpensify** [comment]
@kbecciv and @Santhosh-Sellavel - close out this issue, since it's not reproducible anymore?

**Comment by Santhosh-Sellavel** [comment]
Yes we are good to close this if not reproducible this week also.

**Comment by stitesExpensify** [comment]
not overdue, waiting to see if it is still not reproducible

**Comment by mvtglobally** [comment]
team saw the following this round 

https://user-images.githubusercontent.com/43995119/182288946-3238ebe4-2046-4456-9e99-d77f9fec93b2.mp4

**Comment by jliexpensify** [comment]
@stitesExpensify and @Santhosh-Sellavel - it looks like the keyboard is encroaching on the New Password field: however - the Save button isn't hidden, and the New Password field isn't completely hidden either and you can still see what's in the field. 

Would you say that this is an issue? If so, we could update the title to "New Password field is partially hidden by keyboard popping up".

**Comment by stitesExpensify** [comment]
Yes, I think this should be fixed. If you could update the title and OP description that would be great. Thanks!

**Comment by jliexpensify** [comment]
I've had to take some unexpected time off, so assigning to Kadie (my leave buddy) to manage!

**Comment by kadiealexander** [comment]
Old Upwork post expired so I've created a new one and doubled the price.
https://www.upwork.com/ab/applicants/1556866686745866240/job-details
https://www.upwork.com/jobs/~01a8d7f156f66a3afa

**Comment by Faizan-omer** [comment]
Hi seems like this is an issue with android devices I am trying to access the run the web version of the project on my local system but I keep getting a 403 forbidden error from the server can you please guide me on how to set it up so that I may debug the issue

**Comment by Faizan-omer** [comment]
It seems like the issue is only happening on android devices I am trying to login with my local system but I keep getting 403 forbidden response from the server let me know how to access the server so I can debug the issue.

**Comment by Santhosh-Sellavel** [comment]
I'm not sure what error you are facing, please ask in #expensify-open-source with more details so someone might help you.

A common issue is at times emulator might have some cache issues like time, try resetting or creating a new emulator and check. Try in a real device if have one, thanks!

**Comment by kadiealexander** [comment]
Price doubled again!

**Comment by kadiealexander** [comment]
Job post expired so I've created a new one:

https://www.upwork.com/jobs/~015e9429d8955b230c
https://www.upwork.com/ab/applicants/1562790642649567232/job-details

**Comment by ntdiary** [comment]
# Proposal
## Solution
I think we can be consistent with the expected behavior in this [slack chat](https://expensify.slack.com/archives/C01GTK53T8Q/p1661281246518809).
> all bottom-docked buttons only stay docked to the bottom of the view if there is no scroll above them.
> ...
> as the parent container shrinks (because now a keyboard is in view) then the content area will use a minimum height of its content

Just use `flex-grow` or use [Form.js](https://github.com/Expensify/App/blob/main/src/components/Form.js) to refactor the component like [this issue](https://github.com/Expensify/App/issues/9575). (the final effect is same)
```diff
diff --git a/src/pages/settings/PasswordPage.js b/src/pages/settings/PasswordPage.js
index 69270c2ba..b78a9c0a4 100755
--- a/src/pages/settings/PasswordPage.js
+++ b/src/pages/settings/PasswordPage.js
@@ -159,12 +159,13 @@ class PasswordPage extends Component {
                         <>
                             <ScrollView
                                 style={styles.flex1}
-                                contentContainerStyle={styles.p5}
+                                contentContainerStyle={[styles.p5, styles.flexGrow1]}

                                 // Allow the user to click show password while password input is focused.
                                 // eslint-disable-next-line react/jsx-props-no-multi-spaces
                                 keyboardShouldPersistTaps="always"
                             >
+                                <View style={styles.flexGrow1}>
                                     <Text style={[styles.mb6]}>
                                         {this.props.translate('passwordPage.changingYourPasswordPrompt')}
                                     </Text>
@@ -213,15 +214,14 @@ class PasswordPage extends Component {
                                             {this.props.account.error}
                                         </Text>
                                     )}
-                            </ScrollView>
-                            <FixedFooter style={[styles.flexGrow0]}>
+                                </View>
                                 <Button
                                     success
                                     isLoading={this.props.account.loading}
                                     text={this.props.translate('common.save')}
                                     onPress={this.submit}
                                 />
-                            </FixedFooter>
+                            </ScrollView>
                         </>
                     )}
             </ScreenWrapper>
```
## mobile web
chrome(1080x1920)

https://user-images.githubusercontent.com/8579651/186870441-cb53a68b-ddc2-4c49-ba89-af93123bb0fc.mp4

chrome(1236x2676)

https://user-images.githubusercontent.com/8579651/186870825-376d3832-c5f6-4da4-a41b-a75e46c95053.mp4

safari

https://user-images.githubusercontent.com/8579651/186871298-7198754f-fe7f-4fe9-9a8e-1568527e872c.mp4

## android

https://user-images.githubusercontent.com/8579651/186871066-745bd0c3-0d8f-4daf-b52e-93fd9c384047.mp4

## ios

https://user-images.githubusercontent.com/8579651/186871428-933a6de0-4f08-4c05-a1b1-5b1810202457.mp4

## web

https://user-images.githubusercontent.com/8579651/186871542-33d2bdd8-77b5-4068-8071-e44e63d025a7.mp4

## desktop

https://user-images.githubusercontent.com/8579651/186871865-64fb0e99-6674-488e-92a5-527689e1544f.mp4

**Comment by Santhosh-Sellavel** [comment]
@Expensify/design Any thoughts on this?

**Comment by shawnborton** [comment]
For @ntdiary 's [proposal above](https://github.com/Expensify/App/issues/9620#issuecomment-1228277931), can we see it on a mobile device that is quite small, like an iPhone SE?

**Comment by ntdiary** [comment]
iPhone SE simulator:

https://user-images.githubusercontent.com/8579651/187143499-24d217c3-2a98-45c9-a4a9-cc050e790e87.mp4

**Comment by stitesExpensify** [comment]
Not overdue, there is active discussion

**Comment by shawnborton** [comment]
The behavior from the iPhone SE video feels strange - why is the green button so far away from the bottom input?

**Comment by Santhosh-Sellavel** [comment]
> The behavior from the iPhone SE video feels strange - why is the green button so far away from the bottom input? 

@shawnborton That's common issue that occurs in mWeb Safari it exists it all pages and forms, and we have separate issue for that.

**Comment by shawnborton** [comment]
Sounds good, thanks for confirming.

**Comment by Santhosh-Sellavel** [comment]
@stitesExpensify 

@ntdiary [proposal](https://github.com/Expensify/App/issues/9620#issuecomment-1228277931) looks good to me! 

What we prefer though form migration or `flex-grow` styling?

cc:@shawnborton

**Comment by shawnborton** [comment]
I think for me, I generally prefer what @wildan-m shared in [this Slack post here](https://expensify.slack.com/archives/C01GTK53T8Q/p1661315954700859?thread_ts=1661281246.518809&cid=C01GTK53T8Q) as the standard behavior for these kinds of forms. 

So maybe it's just a matter of making sure we standardize on that behavior and document it as the expected behavior for bottom-docked buttons?

**Comment by ntdiary** [comment]
Hi @Santhosh-Sellavel , I've updated form migration detail to the [proposal](https://github.com/Expensify/App/issues/9620#issuecomment-1228277931) if needed. 🙂
this is the [commit](https://github.com/ntdiary/App/commit/a3632d42fff939d06d5915910a5d6fbb69fe5607). 
In addition, the keyboard flickers when navigating back on android. (main branch)
This commit is also trying to fix this issue.

before:

https://user-images.githubusercontent.com/8579651/188624744-db45d55e-b288-46df-a288-401cd6bda389.mp4

after:

https://user-images.githubusercontent.com/8579651/188625623-f4e43f8d-cf6e-48ca-bd81-08a52eb92492.mp4

**Comment by Santhosh-Sellavel** [comment]
@stitesExpensify  
I am happy to move forward with @ntdiary's proposal of migrating to Form!

> "New Password save button is partially covered by keyboard".

This would be a common issue that would be occurring only in `mWeb`, it's common to all the forms that should be a separate issue. Also, I believe it would be handled [here](https://github.com/Expensify/App/issues/10273)

Regarding the keyboard flickering issue, it should be a separate issue. 

C+ Reviewed
🎀 👀 🎀 
cc: @shawnborton

**Comment by stitesExpensify** [comment]
Hm so we've been talking about a lot of these things internally because these issues have been popping up so much. @roryabraham is this another one you think should be held for now?

**Comment by mallenexpensify** [comment]
We're putting all keyboard-related issues on hold for a bit while we discuss a more holistic approach to fixing them (also https://github.com/Expensify/App/issues/9542 and https://github.com/Expensify/App/issues/9252).

@ntdiary it's clear you've invested a lot of time into this issue, your proposal is extremely thorough and well-outlined. You will be compensated for your work, before discussing the amount, we want to see if the best path forward is for you to complete the issue. If so, we'd take off hold then you'd finish then get compensated 7 days after the PR hits production, assuming no regressions. We're discussing next steps internally and will have another update on the hold timeline soon (hopefully tomorrow)

**Comment by kadiealexander** [comment]
Just adding a note that I'm OOO for the next 9 days until September 21st. As this is on hold I won't reassign it, but if you need a contributor manager to do anything before that point please reapply the `external` label.

**Comment by kadiealexander** [comment]
@mallenexpensify any update on your post above?

**Comment by stitesExpensify** [comment]
Bump :)

**Comment by roryabraham** [comment]
I think @tgolen has started working on the keyboard-related bugs and improvements tracking issue.

**Comment by shawnborton** [comment]
Not overdue

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by kadiealexander** [comment]
@roryabraham Both applause and I are unable to reproduce this, and it's been on hold forever. Do you think we should close it for now?

**Comment by tgolen** [comment]
I think we should just close this, yes.

**Comment by stitesExpensify** [comment]
Yep! Closing it out.
